docs/nodejs_dev_guide.rst

```diff
@@ -215,148 +199,359 @@ Add app.js and update package.json's scripts
+            VIRTUAL_PORT: 3000
+            HTTPS_METHOD: noredirect # support both http and https
+          env_file:
+            - .env-common
+            - .env-dev
+          ports:
+            - "3000"
+            - "5858"
+          volumes:
+            - .:/opt/app
+          restart: unless-stopped
+          # to get this work with https://github.com/jwilder/nginx-proxy
+          # related: https://github.com/jwilder/nginx-proxy/issues/305
+          network_mode: bridge
 
-      ADD package.json $APP/
 
-      WORKDIR $APP
+- Create ``run-dev.sh`` with the following content:
 
-      RUN npm install
+  ..  code-block:: bash
 
-      VOLUME $APP/node_modules
+      #!/bin/bash
 
-- Create ``docker-compose.yml`` file within the ``hello-world-nodejs`` directory with the following
-  content:
+      yarn
 
-  ..  code-block:: yaml
+      yarn run dev
 
-      version: '2'
 
-      services:
-        dev:
-          build:
-            context: .
-            dockerfile: Dockerfile-dev
-          image: teracy/hello-world-nodejs:dev_develop
-          command: bash -c "npm run start:dev"
-          volumes:
-            - .:/opt/app
-          ports:
-            - 3000
-          restart: always
-          # nginx-proxy
-          environment:
-            - VIRTUAL_HOST=hello-d.teracy.dev
-          # to get this work with https://github.com/jwilder/nginx-proxy
-          # related: https://github.com/jwilder/nginx-proxy/issues/305
-          network_mode: bridge
 
-- Create ``Dockerfile`` file within the ``hello-world-nodejs`` directory with the following content:
+- Create ``.env-common`` and ``.env-dev`` files to set environment variables when required.
+
+
+The changes should be like this:
+https://github.com/acme101/nodejs-hello-world/commit/b0a7ac4e95898ecd12651830917a5f4db9561420
+
+
+Run on dev mode
+---------------
+
+Open a new terminal window, ``ssh`` into the ``teracy-dev`` VM to execute the following commands:
+
+..  code-block:: bash
+
+    $ vagrant ssh
+    $ ws
+    $ cd nodejs-hello-world
+    $ docker-compose up -d dev && docker-compose logs -f dev
+
+
+After that, open:
+
+- http://dev.nodejs.teracy.dev or https://dev.nodejs.teracy.dev on your host browser to see the app
+  on the dev mode.
+- Check out the VM's :ref:`basic_usage-ip_address` and on any device within your LAN,
+  open http://dev.nodejs.<vm_ip>.xip.io or https://dev.nodejs.<vm_ip>.xip.io to see the web app.
+
+
+2 way sync for node_modules
+---------------------------
+
+By default, we disable ``node_modules`` sync. To have 2 way sync, we need to configure it.
+
+- Create ``node_modules`` directory from the host terminal window:
+
+  ..  code-block:: bash
+
+      $ cd ~/teracy-dev/workspace/nodejs-hello-world
+      $ mkdir node_modules
+      $ touch node_modules/.gitkeep
+
+- Add gitignore for node_modules content, except .gitkeep to keep the empty directory, this is required
```

Comment:
Add ``gitignore`` for the ``node_modules`` content, except ``.gitkeep`` to keep the empty directory. This is required
by the configuration below.